core/Db.php

```diff
@@ -466,63 +467,13 @@ public static function deleteAllRows($table, $where, $orderBy, $maxRowsPerQuery
      *                             Table names must be prefixed (see {@link Piwik\Common::prefixTable()}).
      * @param bool $force If true, the `OPTIMIZE TABLE` query will be run even if InnoDB tables are being used.
      * @return bool
+     * @deprecated  will be removed in Matomo 6
+     *              use Schema::getInstance()->optimizeTables() instead
      */
     public static function optimizeTables($tables, $force = false)
     {
-        $optimize = Config::getInstance()->General['enable_sql_optimize_queries'];
-
-        if (
-            empty($optimize)
-            && !$force
-        ) {
-            return false;
-        }
-
-        if (empty($tables)) {
-            return false;
-        }
-
-        if (!is_array($tables)) {
-            $tables = array($tables);
-        }
-
-        if (
-            !self::isOptimizeInnoDBSupported()
-            && !$force
-        ) {
-            // filter out all InnoDB tables
-            $myisamDbTables = array();
-            foreach (self::getTableStatus() as $row) {
-                if (
-                    strtolower($row['Engine']) == 'myisam'
-                    && in_array($row['Name'], $tables)
-                ) {
-                    $myisamDbTables[] = $row['Name'];
-                }
-            }
-
-            $tables = $myisamDbTables;
-        }
-
-        if (empty($tables)) {
-            return false;
-        }
-
-        // optimize the tables
-        $success = true;
-        foreach ($tables as &$t) {
-            $ok = self::query('OPTIMIZE TABLE ' . $t);
-            if (!$ok) {
-                $success = false;
-            }
-        }
-
-        return $success;
-    }
-
-    private static function getTableStatus()
-    {
-        return Db::fetchAll("SHOW TABLE STATUS");
+        $tables = !is_array($tables) ? [$tables] : $tables;
```

Comment:
Are we reasonably sure noone passed in an empty non-array value for `$tables`?

The previous `empty($tables)` check was done before the `is_array()` check, potentially catching that edge case. I would deem the risk of that case to be negligible though :thinking: